xarray/core/groupby.py

```diff
@@ -201,7 +201,7 @@ def _unique_and_monotonic(group):
     if isinstance(group, _DummyGroup):
         return True
     index = safe_cast_to_index(group)
-    return index.is_unique and index.is_monotonic
+    return index.is_unique and index.is_monotonic_increasing
```

Comment:
@dcherian do you know if `is_monotonic_increasing` correct here or should that be `(is_monotonic_increasing or is_monotonic_decreasing)`.

(Note that `pd.Index.is_monotonic` is an alias for `pd.Index.is_monotonic_increasing`).

Reply:
`(is_monotonic_increasing or is_monotonic_decreasing)` I think this is the intent (see lines 355-362 below where we avoid calling pd.factorize when the indices are unique and monotonic) but I doubt it's tested. I think we just end up using the "slow" path with more copies for monotonically decreasing group variables.

Reply:
Opened #6220

Reply:
Thanks! Let's merge this as is then.